.travis.yml

```diff
@@ -108,13 +108,20 @@ jobs:
       os: linux
       language: python
       env: [ TEST_SUITE=build, 'SPEC=mpich' ]
+    - stage: 'docker build'
+      sudo: required
+      os: linux
+      language: generic
+      env: TEST_SUITE=docker
   allow_failures:
     - dist: xenial
 
 stages:
   - 'style checks'
   - 'unit tests + documentation'
   - 'build tests'
+  - name: 'docker build'
+    if: type IN (api, cron, push)
```

Comment:
Do we want this to happen on `api` or `cron` builds?  `push` should be enough to handle each new commit on `develop`, right?  I think doing this on `api` is going to be bad, as I think that will only happen when we're testing something.  I could be wrong...

Reply:
I'm not certain how `api` and `cron` builds work in this context.  The particular change was in response to [this](https://github.com/spack/spack/pull/9329#discussion_r224696472) suggestion from the earlier review.  I'll list some alternatives below and you can just thumbs-up the one you prefer.

Reply:
Predicate on the branch name?

```Yaml
- name: 'docker build'
  if: branch = develop
```

Reply:
Predicate on the type (push only)?

```Yaml
- name: 'docker build'
  if: type = push
```

Reply:
Or, leave it as-is; predicate on the type (api, cron, and push)?

```Yaml
- name: 'docker build'
  if: type IN (api, cron, push)
```

Reply:
I think we just want pushes to develop and master for now.  I'll update it.